src/bst.py

```diff
@@ -183,3 +184,37 @@ def breadth_first(self):
                 pending_list.append(cur.left)
             if cur.right:
                 pending_list.append(cur.right)
+
+    def _search(self, val):
+        cur = self.root
+        if not cur:
+            raise ValueError("Val not in BST")
+        if cur.val == val:
+            return cur
+        while True:
+            if val > cur.val and cur.right is not None:
+                parent = cur
+                cur = cur.right
+                cur.parent = parent
+                if cur.val == val:
+                    return cur
+            elif val < cur.val and cur.left is not None:
+                parent = cur
+                cur = cur.left
+                cur.parent = parent
+                if cur.val == val:
+                    return cur
+            else:
+                break
+
+    def delete(self, val):
+        cur = self._search(val)
+        if cur is None:
+            raise ValueError("Val not in BST")
```

Comment:
Per the assignment specs, your `delete` method should be a no-op (no operation) when the value is not in the tree, rather than raising an error.